app/models/order_detail.rb

```diff
@@ -771,6 +747,34 @@ def problem_order?
     problem
   end
 
+  def can_dispute?
+    in_review? && dispute_at.blank?
+  end
+
+  def in_dispute?
+    dispute_at.present? && dispute_resolved_at.nil? && !canceled?
+  end
+
+  def in_review?
+    reviewed_at.try(:future?) && !in_dispute?
+  end
+
+  def reviewed?
+    reviewed_at.try(:past?) && !in_dispute?
+  end
+
+  def ready_for_statement?
+    reviewed? && statement_id.blank? && Account.config.using_statements?(account.type)
+  end
+
+  def ready_for_journal?
+    reviewed? && journal_id.blank? && Account.config.using_journal?(account.type)
+  end
+
+  def awaiting_payment?
```

Comment:
Correct, while it has slightly different underlying logic, this method is analogous to `in_open_journal?`